compiler/rustc_interface/src/passes.rs

```diff
@@ -647,6 +666,10 @@ pub fn write_dep_info(tcx: TyCtxt<'_>) {
     // the side-effect of providing a complete set of all
     // accessed files and env vars.
     let _ = tcx.resolver_for_lowering();
+    // Similarly, allow codegen and linking passes to state which environment
+    // variables they depend on, as those passes are run after this pass, but
+    // we'll need the information when emitting dependency info to Cargo.
+    rustc_codegen_ssa::register_environment_variables(tcx);
```

Comment:
I don't think it makes sense to add this until we are actually doing something there.